src/goto-instrument/contracts/dynamic-frames/dfcc_root_object.cpp

```diff
@@ -0,0 +1,310 @@
+/*******************************************************************\
+
+Module: Dynamic frame condition checking
+
+Author: Remi Delmas, delmasrd@amazon.com
+
+Date: March 2023
+
+\*******************************************************************/
+
+#include "dfcc_root_object.h"
+
+#include <util/byte_operators.h>
+#include <util/cprover_prefix.h>
+#include <util/expr.h>
+#include <util/expr_util.h>
+#include <util/message.h>
+#include <util/namespace.h>
+#include <util/pointer_expr.h>
+#include <util/std_code.h>
+
+#include <langapi/language_util.h>
+
+/// Translates object slice expressions found in assigns clause targets to
+/// dereference expressions, so that object_descriptor_exprt can be used to
+/// compute the base object expression.
+static const exprt slice_op_to_deref(const exprt &expr)
+{
+  if(can_cast_expr<side_effect_expr_function_callt>(expr))
+  {
+    auto function_call_expr = to_side_effect_expr_function_call(expr);
+    auto function_expr = function_call_expr.function();
+    INVARIANT(
+      function_expr.id() == ID_symbol,
+      "no function pointer calls loop assigns clause targets");
```

Comment:
Missing "in" (no function pointer calls *in* loop ...)

Reply:
fixed